pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -252,35 +254,54 @@ cpdef inline object precision_from_unit(object unit):
         int64_t m
         int p
 
-    if unit == 'Y':
-        m = 1000000000L * 31556952
+    if unit is None:
+        m = 1L
+        p = 0
+        return m, p
+
+    unit, stride = _base_and_stride(unit)
+
+    # Normalize old or lowercase codes to standard offset aliases.
+    if unit in ['min', 'm']:
+        unit = 'T'
+    elif unit == 'ns':
+        unit = 'N'
+    elif unit == 'ms':
+        unit = 'L'
+    elif unit == 'us':
+        unit = 'U'
+
```

Comment:
make this an else

Reply:
I don't understand.  I actually want "unit" to be unchanged if not one of the tested values so what would an "else" do?  Wouldn't it force all "unit" values to be "U"?  I have in the past used a dictionary instead to do this same thing if that is a better and more readable approach.

Reply:
```
else:
    unit = unit.upper()
```